### PR TITLE
c/clang:full ast dump

### PR DESCRIPTION
**Pull request by luoliwoshang**
modify some cursor kind via https://github.com/goplus/llgo/pull/667

current dump content 
```
ClassDecl: INIReader (Type: INIReader)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:42:7)
    CXXAccessSpecifier:  public
(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:43:3)
    CXXConstructor: INIReader (Symbol: __ZN9INIReaderC1ERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEE)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:46:22)
      attribute(visibility): default(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:46:5)
      ParmDecl: filename (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:46:51)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:46:38)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:46:43)
    CXXConstructor: INIReader (Symbol: __ZN9INIReaderC1EPKcm)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:50:22)
      attribute(visibility): default(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:50:5)
      ParmDecl: buffer (Type: const char *)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:50:44)
      ParmDecl: buffer_size (Type: size_t)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:50:59)
        TypeRef: size_t (Type: size_t)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:50:52)
    CXXMethod: ParseError (Symbol: __ZNK9INIReader10ParseErrorEv)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:54:17)
      attribute(visibility): default(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:54:5)
    CXXMethod: Get (Symbol: __ZNK9INIReader3GetERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEES8_S8_)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:57:25)
      attribute(visibility): default(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:57:5)
      NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:57:13)
      TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:57:18)
      ParmDecl: section (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:57:48)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:57:35)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:57:40)
      ParmDecl: name (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:57:76)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:57:63)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:57:68)
      ParmDecl: default_value (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:58:48)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:58:35)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:58:40)
    CXXMethod: GetString (Symbol: __ZNK9INIReader9GetStringERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEES8_S8_)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:62:25)
      attribute(visibility): default(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:62:5)
      NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:62:13)
      TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:62:18)
      ParmDecl: section (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:62:54)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:62:41)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:62:46)
      ParmDecl: name (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:62:82)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:62:69)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:62:74)
      ParmDecl: default_value (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:63:54)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:63:41)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:63:46)
    CXXMethod: GetInteger (Symbol: __ZNK9INIReader10GetIntegerERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEES8_l)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:67:18)
      attribute(visibility): default(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:67:5)
      ParmDecl: section (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:67:48)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:67:35)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:67:40)
      ParmDecl: name (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:67:76)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:67:63)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:67:68)
      ParmDecl: default_value (Type: long)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:67:87)
    CXXMethod: GetInteger64 (Symbol: __ZNK9INIReader12GetInteger64ERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEES8_x)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:71:21)
      attribute(visibility): default(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:71:5)
      TypeRef: int64_t (Type: int64_t)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:71:13)
      ParmDecl: section (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:71:53)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:71:40)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:71:45)
      ParmDecl: name (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:71:81)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:71:68)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:71:73)
      ParmDecl: default_value (Type: int64_t)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:71:95)
        TypeRef: int64_t (Type: int64_t)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:71:87)
    CXXMethod: GetUnsigned (Symbol: __ZNK9INIReader11GetUnsignedERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEES8_m)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:75:27)
      attribute(visibility): default(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:75:5)
      ParmDecl: section (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:75:58)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:75:45)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:75:50)
      ParmDecl: name (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:75:86)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:75:73)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:75:78)
      ParmDecl: default_value (Type: unsigned long)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:76:53)
    CXXMethod: GetUnsigned64 (Symbol: __ZNK9INIReader13GetUnsigned64ERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEES8_y)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:80:22)
      attribute(visibility): default(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:80:5)
      TypeRef: uint64_t (Type: uint64_t)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:80:13)
      ParmDecl: section (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:80:55)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:80:42)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:80:47)
      ParmDecl: name (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:80:83)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:80:70)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:80:75)
      ParmDecl: default_value (Type: uint64_t)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:80:98)
        TypeRef: uint64_t (Type: uint64_t)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:80:89)
    CXXMethod: GetReal (Symbol: __ZNK9INIReader7GetRealERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEES8_d)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:85:20)
      attribute(visibility): default(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:85:5)
      ParmDecl: section (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:85:47)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:85:34)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:85:39)
      ParmDecl: name (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:85:75)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:85:62)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:85:67)
      ParmDecl: default_value (Type: double)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:85:88)
    CXXMethod: GetBoolean (Symbol: __ZNK9INIReader10GetBooleanERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEES8_b)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:90:18)
      attribute(visibility): default(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:90:5)
      ParmDecl: section (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:90:48)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:90:35)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:90:40)
      ParmDecl: name (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:90:76)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:90:63)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:90:68)
      ParmDecl: default_value (Type: bool)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:90:87)
    CXXMethod: HasSection (Symbol: __ZNK9INIReader10HasSectionERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEE)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:94:18)
      attribute(visibility): default(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:94:5)
      ParmDecl: section (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:94:48)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:94:35)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:94:40)
    CXXMethod: HasValue (Symbol: __ZNK9INIReader8HasValueERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEES8_)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:97:18)
      attribute(visibility): default(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:97:5)
      ParmDecl: section (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:97:46)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:97:33)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:97:38)
      ParmDecl: name (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:97:74)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:97:61)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:97:66)
    CXXAccessSpecifier:  private
(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:99:3)
    FieldDecl: _error (Type: int)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:100:9)
    FieldDecl: _values (Type: int)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:101:40)
    CXXMethod: MakeKey (Symbol: __ZN9INIReader7MakeKeyERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEES8_)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:102:24)
      NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:102:12)
      TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:102:17)
      ParmDecl: section (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:102:51)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:102:38)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:102:43)
      ParmDecl: name (Type: const std::string &)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:102:79)
        NamespaceRef: std(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:102:66)
        TypeRef: std::string (Type: std::string)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:102:71)
    CXXMethod: ValueHandler (Symbol: __ZN9INIReader12ValueHandlerEPvPKcS2_S2_)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:103:16)
      ParmDecl: user (Type: void *)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:103:35)
      ParmDecl: section (Type: const char *)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:103:53)
      ParmDecl: name (Type: const char *)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:103:74)
      ParmDecl: value (Type: const char *)(Loc:/opt/homebrew/Cellar/inih/58/include/INIReader.h:103:92)
```

### CODE REVIEW

File: c/clang/_demo/castdump/astdump.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package main
 import (
```
Comment (xushiwei) [inline c/clang/_demo/castdump/astdump.go]: move to `chore/castdump`.

Comment (luoliwoshang) [reply c/clang/_demo/castdump/astdump.go]: > move to `chore/castdump`.

In progress, also do we need to dump it to json via cjson

**Comment by xushiwei** [comment]
sorry. move to `chore/_xtool/castdump`

**Comment by luoliwoshang** [comment]
> sorry. move to `chore/_xtool/castdump`

fixed

**Comment by luoliwoshang** [comment]
> sorry. move to `chore/_xtool/castdump`

moved

**Pull request merged**